### ISSUE TITLE
formatter deviation: line break not occurring on assignments where LHS goes over line-length

### ISSUE THREAD

**Issue by jonathan-hill-visasq** [astral-sh/ruff#11820]
I'm seeing a case where black is expanding an assignment but ruff is not - even re-writing the black formatted result.

To keep it concise, this example uses an ultra-short line-length of 14:

Input
```
long_variable_name = 1234
```

Black:
```
long_variable_name = (
    1234
)
```

Ruff:
```
long_variable_name = 1234

```

This seems to occur only when the name of the variable assigned to, is longer than line-length (so expanding the line won't bring it within the allowed length).

I can't find this in the documented deviations or other issues at present, and this doesn't seem to be a duplicate of another issue.

**Comment by MichaReiser** [comment]
Yes, I can confirm that #8940 would fix the deviation. The problem is, that the PR has a few regressions around call expression formatting, which is why the PR was abandoned. So fixing this requires some more work and is also something that I don't think we can fix outside of a new formatter style guide as it would change how already formatted code gets formatted.
